saleor/checkout/utils.py

```diff
@@ -1024,47 +1011,49 @@ def create_line_for_order(
     if translated_product_name == product_name:
         translated_product_name = ""
 
+    total_line_price = calculate_checkout_line_total(checkout_line, discounts)
+    unit_price = total_line_price / checkout_line.quantity
     line = OrderLine(
         product_name=product_name,
         translated_product_name=translated_product_name,
         product_sku=variant.sku,
         is_shipping_required=variant.is_shipping_required(),
         quantity=quantity,
         variant=variant,
-        unit_price=variant.get_price(discounts, taxes),
-        tax_rate=get_tax_rate_by_name(variant.product.tax_rate, taxes),
+        unit_price=unit_price,
+        tax_rate=unit_price.tax / unit_price.net,
     )
 
     return line
 
 
-def prepare_order_data(
-    *, checkout: Checkout, tracking_code: str, discounts, taxes
-) -> dict:
+def prepare_order_data(*, checkout: Checkout, tracking_code: str, discounts) -> dict:
     """
     Runs checks and returns all the data from a given checkout to create an order.
 
     :raises NotApplicable InsufficientStock:
     """
     order_data = {}
 
-    order_data.update(_process_shipping_data_for_order(checkout, taxes))
+    total = (
+        calculate_checkout_total(checkout=checkout, discounts=discounts)
+        - checkout.get_total_gift_cards_balance()
+    )
+    total = max(total, ZERO_TAXED_MONEY)
```

Comment:
I've tried to replace all `ZERO_` fields with a function that takes currency as an arg and returns `TaxedMoney(0, currency)` but it requires an additional amount of work: currency field for some models, logic to fill up these fields, tests. I am worrying that this will delay this PR even more and we will move out of the scope of given pproblem. I would like to introduce multicurrency in separate PR to keep consistency and solve one problem  at a time. 
@patrys  Do you agree with that?